mdtraj/geometry/src/sasa.cpp

```diff
@@ -63,11 +66,16 @@
  */
 static void asa_frame(const float* frame, const int n_atoms, const float* atom_radii,
                       const float* sphere_points, const int n_sphere_points,
-                      int* neighbor_indices, float* centered_sphere_points, float* areas)
+                      int* neighbor_indices, float* centered_sphere_points, const int* atom_selection_mask, float* areas)
 {
     float constant = 4.0 * M_PI / n_sphere_points;
 
     for (int i = 0; i < n_atoms; i++) {
+        // Skip atom if not in selection
```

Comment:
Would it be better to just loop over the selection, rather than looping over all the atoms and skipping?

Reply:
absolutely, I just don't really know any c++ lol

Reply:
I would:
1. Don't pass in a mask array of booleans, just pas in the list of atom indicies (pointer + length)
2. Make sure to validate that the atom indices are in-bounds.
3. Instead of `for (int i = 0; i < n_atoms; i++)` to loop over the atoms `i`, do `for (int ii = 0; ii < n_atom_indices; ii++) {... i = atom_indices[ii]` to loop over the atoms.

Reply:
Done! Only took me three years!